Mathlib/LinearAlgebra/Basis/Flag.lean

```diff
@@ -63,6 +63,15 @@ theorem isChain_range_flag (b : Basis (Fin n) R M) : IsChain (· ≤ ·) (range
 theorem flag_strictMono [Nontrivial R] (b : Basis (Fin n) R M) : StrictMono b.flag :=
   Fin.strictMono_iff_lt_succ.2 fun _ ↦ by simp [flag_succ]
 
+theorem flag_span_succ {n : ℕ} (b : Basis (Fin n) R M) (k : Fin n) :
```

Comment:
How does this relate to `flag_succ`? Can this lemma be replaced by that?

Everything after the second line of the proof seems like it should be a lemma about `Fin.castSucc`.

Also the name should be something like `flag_sup_span_single`.

Are you silently coercing `k` to `Fin (n+1)` in the statement? Do we allow that in the library?

Reply:
Thank you for the sharp eyes. Indeed, I hadn't seen `flag_succ`.
This lemma differs in two inessential ways (one would need to flip the order on the lhs, and then lhs and rhs in this lemma) and one substantial way (it takes `k.succ` instead of `k.castSucc`).

The first two are easy to fix; the last one requires deeper adaptations in sphere-eversion than I could do quickly. (I asked on zulip.) I'm not sure if this lemma carries its weight. What do you think?

Reply:
To answer your question: yes, there is a silent coercion happening. (I'm not sure about mathlib's policy on this.)

Reply:
I tried again: sphere-eversion can just use `flag_succ`. I'm happy to close this PR.